### PR TITLE
Feature/cvsb 5211

### PR DESCRIPTION
**Pull request by bsiddappa**
Please review the changes done for lambda to lambda call.
Note that I will continue to work on mocks using 1921 ticket and will raise a PR once it is done. These have been tested on AWS and working fine for all cert types.
I have PR raised on doc-gen(mot service) repo too. Kindly review the same in parallel with this.

### CODE REVIEW

File: src/services/CertificateGenerationService.ts [hunk 1]
```diff
@@ -49,42 +49,58 @@ class CertificateGenerationService {
...
+            }),
         };
 
-        return rp(reqParams)
-        .then((response: string) => {
-            const responseBuffer: Buffer = Buffer.from(response, "base64");
+        return this.lambdaClient.invoke(invokeParams)
+            .then((response: PromiseResult<Lambda.Types.InvocationResponse, AWSError>) => {
+                const res: string = JSON.stringify(response);
+                console.log(`Output Response: ${res}`);
```
Comment (bsiddappa) [inline src/services/CertificateGenerationService.ts]: These console logs are temporary as i am trying to retrieve the output for mocks. Will revert these later.

File: src/services/CertificateGenerationService.ts [hunk 2]
```diff
@@ -49,42 +49,58 @@ class CertificateGenerationService {
      */
     public async generateCertificate(testResult: any): Promise<IGeneratedCertificateResponse> {
         const config: IMOTConfig = this.config.getMOTConfig();
+        const iConfig: IInvokeConfig = this.config.getInvokeConfig();
         const testType: any = testResult.testTypes;
-        const payload: any = await this.generatePayload(testResult);
+        const payload: any = JSON.stringify(await this.generatePayload(testResult));
```
Comment (KHodderDVSA) [inline src/services/CertificateGenerationService.ts]: can be a string type now. Should it be, though?

Comment (bsiddappa) [reply src/services/CertificateGenerationService.ts]: Agreed, will change that.

File: src/services/CertificateGenerationService.ts [hunk 3]
```diff
@@ -49,42 +49,58 @@ class CertificateGenerationService {
...
+                    fileFormat: "pdf",
+                    fileName: `${testResult.testResultId}_${testResult.vin}_${testResult.order.current}.pdf`,
+                    fileSize: responseBuffer.byteLength.toString(),
+                    certificate: responseBuffer,
+                    certificateOrder: testResult.order,
+                    email: testResult.testerEmailAddress
+                };
+            })
+            .catch((error: AWSError | Error) => {
+                console.log(error);
```
Comment (akothiyal-deloitte) [inline src/services/CertificateGenerationService.ts]: shouldn't this be console.error ? with both the error and custom msg being logged?

Comment (bsiddappa) [reply src/services/CertificateGenerationService.ts]: I see console.log been used everywhere. Kevin, i remember you or someone from team mentioning that console.error was not logging on AWS and so we used console.log. Can you confirm on this? I see console.log used elsewhere in this file, kindly confirm.

Comment (KHodderDVSA) [reply src/services/CertificateGenerationService.ts]: it makes no effective difference. Cloudwatch handles them the same. Custom error might be a separate conversation - in practise I've found the stack traces point to the relevant source anyway

**Pull request merged**